blog: draft 1.0 release post. (#2453)
electric-sql/electric: website/blog/posts/2025-03-17-electricsql-1.0-released.md

We process millions of requests and transactions each day. With hundreds of thousands of active [shapes](/docs/guides/shapes) and application users.

The chart below is from our [cloud benchmarks](/docs/reference/benchmarks#cloud), showing flat, low latency and memory use scaling sync to 1 million concurrent clients on a single community Postgres:

<figure>
  <ScalabilityChart />
</figure>

## Increasingly powerful

We've been focused on making Electric small and stable. So it scales and just works.

Running real workloads has been key to this, as it's given us a tight feedback loop and flushed out real world bugs and edge cases. At the same time, it's also given us a lot of insight into demand for what to build next. And we have some seriously cool stuff coming. From more expressive partial replication primitives to advanced stream processing, database sync and client-side state management.

More on these soon but to give a sneak preview of some of the work in progress:

- [electric-sql/d2ts](https://github.com/electric-sql/d2ts) differential dataflow in Typescript to allow for flexible, extensible stream processing in front of Electric (in the client or at the cloud edge)

> - [tanstack/optimistic](https://github.com/TanStack/optimistic) collaboration with [TanStack](https://tanstack.com/) to create a new library to simplify managing optimistic state in the client. This is early but the DX looks really promising

Reviewer: ```suggestion
- [TanStack/optimistic](https://github.com/TanStack/optimistic) collaboration with [TanStack](https://tanstack.com/) to create a new library to simplify managing optimistic state in the client. This is early but the DX feels really promising
```

They PascalCase `TanStack` everywhere

The chart below is from our [cloud benchmarks](/docs/reference/benchmarks#cloud), showing flat, low latency and memory use scaling sync to 1 million concurrent clients on a single community Postgres:

<figure>
  <ScalabilityChart />
</figure>

## Increasingly powerful

Our engineering has been focused on making Electric small and stable. So it scales and just works.

Running real workloads has been key to this, as it's given us a tight feedback loop and flushed out real world bugs and edge cases. At the same time, it's also given us a lot of insight into demand for what to build next. And we have some seriously cool stuff coming. From more expressive partial replication primitives to advanced stream processing, database sync and client-side state management.

More on these soon but to give a sneak preview of some of the work in progress:

- [electric-sql/d2ts](https://github.com/electric-sql/d2ts) differential dataflow in Typescript to allow for flexible, extensible stream processing in front of Electric (in the client or at the cloud edge)

> - [KyleAMathews/optimistic](http://github.com/KyleAMathews/optimistic) TanStack-compatible library that simplifies working with synced data and optimistic state in the client

Reviewer: ```suggestion
- [tanstack/optimistic](http://github.com/KyleAMathews/optimistic) We're collaborating with TanStack to create a new library to simplify managing optimistic state in the client. This is early but the DX looks really promising.
```

---
title: "Electric 1.0 released"
description: >-
  With version 1.0 Electric is now in GA. The APIs are stable and the
  sync engine is ready for mission critical, production apps.
excerpt: >-
  With version 1.0 Electric is now in GA. The APIs are stable and the
  sync engine is ready for mission critical, production apps.

> authors: [kyle]

Reviewer: ```suggestion
authors: [james]
```

image: /img/blog/electric-1.0-released/header.jpg
tags: [release]
outline: [2, 3]
post: true
---

> With [version `1.0`](https://github.com/electric-sql/electric/releases), Electric is now in GA. The APIs are stable and the sync engine is ready for mission critical, production apps.

Reviewer: Something something we're really excited

## What is Electric?

Sync makes apps awesome. Electric solves sync.

> [Electric](/) is a Postgres sync engine. It handles the core concerns of [partial replication](/docs/guides/shapes), [fan out](/docs/api/http#caching), and [data delivery](/docs/reference/benchmarks#cloud). So you can build awesome software without rolling your own sync.

Reviewer: I'd include the million user benchmark

Reviewer: Also the user quotes again

## The path to 1.0

In 2024 we [re-built Electric from scratch](/blog/2024/07/17/electric-next) to be simpler, faster, more reliable and more scalable. In December 2024, [we hit BETA](/blog/2024/12/10/electric-beta-release#the-path-to-beta) with production users, [proof of scalability](/docs/reference/benchmarks) and a raft of updated [docs](/docs/intro) and [demos](/demos).

> Since then, we've launched a [managed cloud platform](/product/cloud), run / supported a wide range of production workloads from both open-source and cloud users and merged 200 bug-fix and reliability PRs.

Reviewer: Something about antithesis in here?

## Stable APIs

With the 1.0 release, the core [Electric sync service APIs](/docs/intro) are now stable.

Our policy is now no backwards-incompatible changes in patch or minor releases. You can now build on Electric without tracking the latest changes.

## Production ready

Electric is stable, reliable and scales. It's been stress-tested in production for some time now by companies like [Trigger](https://trigger.dev), [Otto](https://ottogrid.ai) and [IP.world](https://ip.world).

We process millions of requests and transactions each day. With hundreds of thousands of active [shapes](/docs/guides/shapes) and application users.

## Increasingly powerful

Our engineering has been focused on making Electric small and stable. So it scales and just works.

> Running real workloads has been key to this, as it's given us a tight feedback loop and flushed our real world bugs and edge cases. At the same time, it's also given us a lot of insight into demand for what to build next. And we have some seriously cool stuff coming. From more expressive partial replication primitives to advanced stream processing, database sync and client-side state management.

Reviewer 2: ```suggestion
Running real workloads has been key to this, as it's given us a tight feedback loop and flushed out real world bugs and edge cases. At the same time, it's also given us a lot of insight into demand for what to build next. And we have some seriously cool stuff coming. From more expressive partial replication primitives to advanced stream processing, database sync and client-side state management.
```

More on these soon but to give a sneak preview of some of the work in progress:

- [electric-sql/d2ts](https://github.com/electric-sql/d2ts) differential dataflow in Typescript to allow for flexible, extensible stream processing in front of Electric (in the client or at the cloud edge)
- [KyleAMathews/optimistic](http://github.com/KyleAMathews/optimistic) TanStack-compatible library that simplifies working with synced data and optimistic state in the client
- [electric-sql/phoenix_sync](https://github.com/electric-sql/phoenix_sync) Phoenix.Sync library to officially add sync support into the Elixir Phoenix web framework

> - [livestore.dev](https://livestore.dev/getting-started/react-web) highly performant reactive state management solution for web and mobile apps with built in Electric sync

Reviewer: ```suggestion
- [livestore.dev](https://livestore.dev/getting-started/react-web) highly performant reactive state management solution for web and mobile apps with built in support for syncing with Electric.
```
